### ISSUE TITLE
grain modules using tuples affect targeting

### ISSUE THREAD

**Issue by robgott** [saltstack/salt#46826]
### Description of Issue/Question
when a grain module returns a tuple as opposed to a list it can't be targeted by value as expected

for example `osrelease_info` returns a tuple and as a result cannot be targeted based on a value eg `osrelease_info:6`

what is confusing is that a master will match a minion ok however the minion itself will not match, so running `salt -G "osrelease_info:6" grains.get saltversion` will yield `Minion did not return. [No response]` from all of the minions that did actually match for `osrelease_info:6`

changing the core.py grain definition for osrelease_info to return a list fixes the issue for me but possibly osrelease_info was made a tuple for performance reasons - #13822

probably finding out why the minion can't access `:0` on the tuple would be the right way to fix

### Setup
nothing special

### Steps to Reproduce Issue
salt -G "osrelease_info:6" grains.get saltversion
salt-call grains.get osrelease_info:6

### Versions Report
confirmed issue on 2015.5 and 2017.7

```Salt Version:
           Salt: 2017.7.4

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 2.6.1
      docker-py: Not Installed
          gitdb: 2.0.3
      gitpython: 2.1.9
          ioflo: Not Installed
         Jinja2: 2.8.1
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.14 (default, Jan 31 2018, 02:12:13)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.5.0
           RAET: Not Installed
          smmap: 2.0.3
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: redhat 6.9 Santiago
         locale: UTF-8
        machine: x86_64
        release: 2.6.32-696.1.1.el6.x86_64
         system: Linux
        version: Red Hat Enterprise Linux Server 6.9 Santiago
```

**Comment by gtmanfred** [comment]
My guess is this is a `isinstance(match, list)` somewhere that should just have tuple added to it.

**Comment by gtmanfred** [comment]
Yup, here is a PR to fix this behavior.

https://github.com/saltstack/salt/pull/46839

**Comment by robgott** [comment]
thanks for the quick reply @gtmanfred

**Comment by rallytime** [comment]
Closed via #46839
